src/components/transport_manager/src/transport_adapter/threaded_socket_connection.cc

```diff
@@ -231,8 +240,14 @@ void ThreadedSocketConnection::Transmit() {
     return;
   }
 
-  // send data if possible
-  if (!frames_to_send_.empty() && (poll_fds[0].revents | POLLOUT)) {
+  {
+    // Check Frames queue is empty or not
+    sync_primitives::AutoLock auto_lock(frames_to_send_mutex_);
+    is_queue_empty = frames_to_send_.empty();
+  }
```

Comment:
May be add some king of ``IsOutPutQueueEmpty()`` method with locking?
It will be more clear code, no duplication and no such lock scopes.